packages/ember-application/lib/system/application.js

```diff
@@ -693,9 +693,15 @@ var Application = Namespace.extend(DeferredMixin, {
     var namespace = this;
     var initializer;
 
+    var makeGraphValue = function(initializer) {
+      return function() {
+        return initializer.initialize.apply(initializer, arguments);
```

Comment:
@gf3 - Did you test it as I suggested?  Calling a function from an object should set `this` inside the function to the object that it was called from.

```javascript
return initializer.initialize(container, application);
```

Sample JSBin: http://jsbin.com/wefas/1/edit?html,js,output